### PR TITLE
Fetch remote urls

### PR DESCRIPTION
**Pull request by technoweenie**
This adds support for pushing and pulling LFS objects with a raw remote url. This was originally submitting in #1085, and updated for the current master branch. This also adds tests.

### CODE REVIEW

File: config/endpoint.go [hunk 1]
```diff
@@ -37,12 +37,17 @@ func NewEndpointFromCloneURLWithConfig(url string, c *Configuration) Endpoint {
 		return e
 	}
 
+	if strings.HasSuffix(url, "/") {
```
Comment (ttaylorr) [inline config/endpoint.go]: Since we already have `path` imported, would it be possible to use `path.Clean` here?

Comment (technoweenie) [inline config/endpoint.go]: We could, but it does a lot more stuff besides just chomping the trailing `/` that we don't need. These urls either come from the git remote config (`remote.origin.url`), or from custom lfs config (`lfs.url`, `remote.origin.lfsurl`).

File: test/test-fetch.sh [hunk 2]
```diff
@@ -503,6 +503,29 @@ begin_test "fetch --prune"
...
+  cd raw
+  git init
+  git lfs install --local --skip-smudge
+
+  git remote add origin $GITSERVER/test-fetch
+  git pull origin master
+
+  # LFS object not downloaded, pointer in working directory
+  refute_local_object "$contents_oid"
+  ack "$content_oid" a.dat
```
Comment (ttaylorr) [inline test/test-fetch.sh]: I don't think the containers that Travis gives us have `ack` installed.

File: test/test-pull.sh [hunk 3]
```diff
@@ -111,6 +111,34 @@ begin_test "pull"
...
+
+  git remote add origin $GITSERVER/test-pull
+  git pull origin master
+
+  contents="a"
+  contents_oid=$(calc_oid "$contents")
+
+  # LFS object not downloaded, pointer in working directory
+  refute_local_object "$contents_oid"
+  ack "$contents_oid" a.dat
```
Comment (ttaylorr) [inline test/test-pull.sh]: Same here.,

**Comment by ttaylorr** [comment]
After you switch out `ack` for `grep`, I think this is ready to merge. 😄

**Pull request merged**